components/formats-api/src/loci/formats/FormatReader.java

```diff
@@ -1459,7 +1429,16 @@ public void setId(String id) throws FormatException, IOException {
       if (store instanceof OMEXMLMetadata) {
         ((OMEXMLMetadata) store).resolveReferences();
         setupService();
-
+        
+        if (getMetadataOptions().isValidate()) {
+          try {
+            String omexml = service.getOMEXML((MetadataRetrieve)store);
+            service.validateOMEXML(omexml);
```

Comment:
From https://ci.openmicroscopy.org/view/Failing/job/BIOFORMATS-DEV-merge-test_images_good/, `NullPointerException` should also be handled if `omexml` is `null`.